phpBB/report.php

```diff
@@ -18,322 +20,22 @@
-					WHERE topic_id = ' . $report_data['topic_id'] . '
-						OR topic_moved_id = ' . $report_data['topic_id'];
-				$db->sql_query($sql);
-			}
-
-			$lang_return = $user->lang['RETURN_TOPIC'];
-			$lang_success = $user->lang['POST_REPORTED_SUCCESS'];
-
-			$phpbb_notifications->add_notifications('notification.type.report_post', array_merge($report_data, $row, $forum_data, array(
-				'report_text'	=> $report_text,
-			)));
-		}
-		else
-		{
-			$sql = 'UPDATE ' . PRIVMSGS_TABLE . '
-				SET message_reported = 1
-				WHERE msg_id = ' . $pm_id;
-			$db->sql_query($sql);
-
-			$sql_ary = array(
-				'msg_id'		=> $pm_id,
-				'user_id'		=> ANONYMOUS,
-				'author_id'		=> (int) $report_data['author_id'],
-				'pm_deleted'	=> 0,
-				'pm_new'		=> 0,
-				'pm_unread'		=> 0,
-				'pm_replied'	=> 0,
-				'pm_marked'		=> 0,
-				'pm_forwarded'	=> 0,
-				'folder_id'		=> PRIVMSGS_INBOX,
-			);
-
-			$sql = 'INSERT INTO ' . PRIVMSGS_TO_TABLE . ' ' . $db->sql_build_array('INSERT', $sql_ary);
-			$db->sql_query($sql);
-
-			$lang_return = $user->lang['RETURN_PM'];
-			$lang_success = $user->lang['PM_REPORTED_SUCCESS'];
-
-			$phpbb_notifications->add_notifications('notification.type.report_pm', array_merge($report_data, $row, array(
-				'report_text'	=> $report_text,
-				'from_user_id'	=> $report_data['author_id'],
-				'report_id'		=> $report_id,
-			)));
-		}
-
-		meta_refresh(3, $redirect_url);
-
-		$message = $lang_success . '<br /><br />' . sprintf($lang_return, '<a href="' . $redirect_url . '">', '</a>');
-		if ($return_forum_url)
-		{
-			$message .= '<br /><br />' . sprintf($user->lang['RETURN_FORUM'], '<a href="' . $return_forum_url . '">', '</a>');
-		}
-		trigger_error($message);
-	}
-	else if (isset($captcha) && $captcha->is_solved() !== false)
-	{
-		$s_hidden_fields .= build_hidden_fields($captcha->get_hidden_fields());
-	}
-}
-
-// Generate the reasons
-display_reasons($reason_id);
-
-$page_title = ($pm_id) ? $user->lang['REPORT_MESSAGE'] : $user->lang['REPORT_POST'];
-
-if (isset($captcha) && $captcha->is_solved() === false)
-{
-	$template->assign_vars(array(
-		'S_CONFIRM_CODE'	=> true,
-		'CAPTCHA_TEMPLATE'	=> $captcha->get_template(),
-	));
-}
-
-$template->assign_vars(array(
-	'ERROR'				=> (sizeof($error)) ? implode('<br />', $error) : '',
-	'S_REPORT_POST'		=> ($pm_id) ? false : true,
-	'REPORT_TEXT'		=> $report_text,
-	'S_REPORT_ACTION'	=> append_sid("{$phpbb_root_path}report.$phpEx", 'f=' . $forum_id . '&amp;p=' . $post_id . '&amp;pm=' . $pm_id),
-	'S_HIDDEN_FIELDS'	=> (sizeof($s_hidden_fields)) ? $s_hidden_fields : null,
-
-	'S_NOTIFY'			=> $user_notify,
-	'S_CAN_NOTIFY'		=> ($user->data['is_registered']) ? true : false,
-	'S_IN_REPORT'		=> true,
+/** @var \phpbb\controller\helper $controller_helper */
+$controller_helper = $phpbb_container->get('controller.helper');
+$response = new RedirectResponse(
+	$controller_helper->route($redirect_route_name, array(
+		'id'	=> ($pm_id === 0) ? $post_id : $pm_id,
+	),
+	301
 ));
```

Comment:
extra ')' here, on should be before 301 to close the array where the second close the new RedirectResponse()